shenyu-admin/src/main/java/org/apache/shenyu/admin/service/register/ShenyuClientRegisterSofaServiceImpl.java

```diff
@@ -0,0 +1,143 @@
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.admin.service.register;
+
+import org.apache.shenyu.admin.listener.DataChangedEvent;
+import org.apache.shenyu.admin.mapper.MetaDataMapper;
+import org.apache.shenyu.admin.mapper.PluginMapper;
+import org.apache.shenyu.admin.mapper.RuleMapper;
+import org.apache.shenyu.admin.model.dto.SelectorDTO;
+import org.apache.shenyu.admin.model.entity.MetaDataDO;
+import org.apache.shenyu.admin.model.entity.PluginDO;
+import org.apache.shenyu.admin.model.entity.RuleDO;
+import org.apache.shenyu.admin.model.entity.SelectorDO;
+import org.apache.shenyu.admin.service.RuleService;
+import org.apache.shenyu.admin.service.SelectorService;
+import org.apache.shenyu.admin.transfer.MetaDataTransfer;
+import org.apache.shenyu.admin.utils.ShenyuResultMessage;
+import org.apache.shenyu.common.enums.ConfigGroupEnum;
+import org.apache.shenyu.common.enums.DataEventTypeEnum;
+import org.apache.shenyu.common.enums.PluginEnum;
+import org.apache.shenyu.common.utils.UUIDUtils;
+import org.apache.shenyu.register.common.dto.MetaDataRegisterDTO;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.util.Collections;
+import java.util.Objects;
+
+/**
+ * sofa service register.
+ *
+ * @author KevinClair
+ **/
+@Service("sofa")
+public class ShenyuClientRegisterSofaServiceImpl extends AbstractShenyuClientRegisterService {
+
+    private final MetaDataMapper metaDataMapper;
+
+    private final ApplicationEventPublisher eventPublisher;
+
+    private final SelectorService selectorService;
+
+    private final RuleService ruleService;
+
+    private final RuleMapper ruleMapper;
+
+    private final PluginMapper pluginMapper;
+
+    public ShenyuClientRegisterSofaServiceImpl(final MetaDataMapper metaDataMapper,
+                                               final ApplicationEventPublisher eventPublisher,
+                                               final SelectorService selectorService,
+                                               final RuleService ruleService,
+                                               final RuleMapper ruleMapper,
+                                               final PluginMapper pluginMapper) {
+        this.metaDataMapper = metaDataMapper;
+        this.eventPublisher = eventPublisher;
+        this.selectorService = selectorService;
+        this.ruleService = ruleService;
+        this.ruleMapper = ruleMapper;
+        this.pluginMapper = pluginMapper;
+    }
+
+    @Override
+    public synchronized String register(final MetaDataRegisterDTO dto) {
+        MetaDataDO metaDataDO = metaDataMapper.findByPath(dto.getPath());
+        if (Objects.nonNull(metaDataDO)
+                && (!metaDataDO.getMethodName().equals(dto.getMethodName())
+                || !metaDataDO.getServiceName().equals(dto.getServiceName()))) {
+            return "you path already exist!";
+        }
```

Comment:
Can we add this part to `AbstractShenyuClientRegisterService`. Both sofa and tars can use it.